src/main/java/org/assertj/vavr/api/VavrInstanceOfAssertFactories.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2017-2023 the original author or authors.
+ */
+package org.assertj.vavr.api;
+
+import io.vavr.Lazy;
+import io.vavr.collection.Map;
+import io.vavr.collection.Multimap;
+import io.vavr.collection.Seq;
+import io.vavr.collection.Set;
+import io.vavr.control.Either;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import io.vavr.control.Validation;
+import org.assertj.core.api.Assert;
+import org.assertj.core.api.InstanceOfAssertFactory;
+
+/**
+ * Static {@link VavrInstanceOfAssertFactories VavrInstanceOfAssertFactories} for {@link Assert#asInstanceOf(InstanceOfAssertFactory)}.
+ *
+ * @author Eduard Dudar
+ * @since 0.4.4
+ */
+@SuppressWarnings({
+        "rawtypes", // using Class instance
+        "unused", // parameter needed for type inference.
+})
+public interface VavrInstanceOfAssertFactories {
```

Comment:
@edudar Thank you for this PR. Good to have it in the project. Please provide tests for it though. Similar to the assertj project.